doc/tutorial.md

```diff
@@ -587,6 +587,37 @@ sp[3]
 See {doc}`/reference/algorithms/index` for details on graph algorithms
 supported.
 
+## Floating point considerations
+
+Many NetworkX algorithms work with numeric values, such as edge weights. Once your input contains floating point numbers, **all results are inherently approximate** due to the limited precision of floating point arithmetic.
+
+For example:
+
+```{code-cell}
+import networkx as nx
+
+eps = 1e-17
+
+G = nx.DiGraph()
+G.add_edge('A', 'B', weight=0.1)
+G.add_edge('B', 'C', weight=0.1)
+G.add_edge('C', 'D', weight=0.1)
+G.add_edge('A', 'D', weight=0.3 + eps)
+
+path = nx.shortest_path(G, 'A', 'D', weight='weight')
+print(path)
+```
+
+Even though `A -> B -> C -> D` has a total weight of `0.3`, the direct edge `A -> D` with `0.3 + 1e-17` is selected. This demonstrates that algorithms treat floating point numbers as exact, so very small differences can change the outcome.
+
+Floating point arithmetic can produce unintuitive results because certain values, such as `0.1`, cannot be represented exactly in binary. For example:
```

Comment:
```suggestion
Floating point arithmetic can produce unintuitive results because certain values, such as `0.3`, cannot be represented exactly in binary. For example:
```